python/cudf_polars/cudf_polars/experimental/expressions.py

```diff
@@ -0,0 +1,485 @@
+
+
+def evaluate_chunk(
+    df: DataFrame,
+    expr: Expr,
+    children: tuple[Expr, ...],
+    *references: Column,
+) -> Column:
+    """Evaluate the sub-expression of a simple FusedExpr node."""
+    return expr.evaluate(df, mapping=dict(zip(children, references, strict=True)))
+
+
+def evaluate_chunk_multi_agg(
+    df: DataFrame,
+    exprs: Sequence[Expr],
+    children: tuple[Expr, ...],
+    *references: Column,
+) -> tuple[Column, ...]:
+    """Evaluate multiple aggregations."""
+    return tuple(
+        expr.evaluate(df, mapping=dict(zip(children, references, strict=True)))
+        for expr in exprs
+    )
+
+
+def combine_chunks_multi_agg(
+    column_chunks: Sequence[tuple[Column]],
+    combine_aggs: Sequence[Agg],
+    finalize: tuple[plc.DataType, str] | None,
+    name: str,
+) -> Column:
+    """Aggregate Column chunks."""
+    column_chunk_lists = zip(*column_chunks, strict=True)
+
+    combined = [
+        agg.op(
+            Column(
+                plc.concatenate.concatenate([col.obj for col in column_chunk_list]),
+                name=column_chunk_list[0].name,
+            )
+        )
+        for agg, column_chunk_list in zip(combine_aggs, column_chunk_lists, strict=True)
+    ]
+
+    if finalize:
+        # Perform optional BinOp on combined columns
+        dt, op_name = finalize
+        op = getattr(plc.binaryop.BinaryOperator, op_name)
+        col = Column(plc.binaryop.binary_operation(*(c.obj for c in combined), op, dt))
+    else:
+        assert len(combined) == 1
+        col = combined[0]
+
+    return col.rename(name)
+
+
+def shuffle_child(
+    child: IR,
+    on: Expr,
+    named_expr: NamedExpr,
+    child_partition_info: PartitionInfo,
+    config_options: ConfigOptions,
+) -> tuple[IR, MutableMapping[Any, Any]]:
+    """
+    Shuffle the child IR on a single expression.
+
+    Parameters
+    ----------
+    child
+        Child IR we are shuffling.
+    on
+        Single Expr to shuffle on.
+    named_expr
+        Outer NamedExpr needing this shuffle.
+    child_partition_info
+        IR partitioning information.
+    config_options
+        GPU engine configuration options.
+
+    Returns
+    -------
+    shuffled
+        New (shuffled) child. The corresponding DataFrame
+        container will only contain the columns associated
+        with the shuffle keys (``on``).
+    graph
+        Task graph needed to shuffle child.
+    """
+    graph: MutableMapping[Any, Any] = {}
+
+    if child_partition_info.partitioned_on == (
+        named_expr.reconstruct(on),
+    ):  # pragma: no cover
+        # No shuffle necessary
+        return child, graph
+
+    child_partition_info = PartitionInfo(count=child_partition_info.count)
+    shuffle_on = (
+        named_expr.reconstruct(
+            FusedExpr(
+                on.dtype,
+                on,
+                *[c for c in named_expr.value.children if isinstance(c, FusedExpr)],
+            )
+        ),
+    )
```

Comment:
```suggestion
    shuffle_on = [
        named_expr.reconstruct(
            FusedExpr(
                on.dtype,
                on,
                *(c for c in named_expr.value.children if isinstance(c, FusedExpr)),
            )
        )
    ]
```

It's easy to miss that this is a singleton tuple.